packages/network-controller/src/NetworkController.ts

```diff
@@ -1374,6 +1374,10 @@ export class NetworkController extends BaseController<
    * @returns networkClientId of the network configuration with the given chainId
    */
   findNetworkClientIdByChainId(chainId: Hex): NetworkClientId {
+    if (this.state.providerConfig.chainId === chainId) {
+      return this.state.providerConfig.id ?? this.state.providerConfig.type;
```

Comment:
Hmm... does it make sense to fall back to `this.state.providerConfig.type` here? Despite the TypeScript types matching, that is not the same thing as the network client ID. To be specific, `NetworkType` (which is the `type` property is) includes "rpc", but there is no network client registered under "rpc". So if you tried to feed this into `getNetworkClientById` you might get an error.

What about:

``` suggestion
    if (this.state.providerConfig.chainId === chainId && this.state.providerConfig.id !== undefined) {
      return this.state.providerConfig.id;
    }
```

Reply:
Seems like a good idea.  I'm trying to handle the fact that NetworkClientIds seem to come from different fields in custom vs infura.  The suggested guard would prevent the (theoretical, shouldn't happen?) case of `type === NetworkType.rpc` but no `id`

Reply:
Yes, practically it shouldn't happen based on the logic we have in `setActiveNetwork` and is purely theoretical based on the fact that NetworkType is a slightly wider type than NetworkClientId.